core/src/main/java/fj/data/hamt/BitSet.java

```diff
@@ -0,0 +1,195 @@
+ * For example, the BitSet("1011") represents the decimal number 11 and has
+ * indices [3, 0] inclusive where the bit with the lowest value has the lowest
+ * index and is the rightmost bit.
+ *
+ * Created by maperr on 31/05/2016.
+ */
+public final class BitSet {
+
+    public static final int TRUE_BIT = 1;
+    public static final int FALSE_BIT = 0;
+
+    public static final long BASE_LONG = 1L;
+    public static final int MAX_BIT_SIZE = Long.SIZE;
+
+    private final long value;
+
+    private BitSet(final long l) {
+        value = l;
+    }
+
+    public static BitSet empty() {
+        return new BitSet(FALSE_BIT);
+    }
+
+    public static BitSet longBitSet(final long l) {
+        return new BitSet(l);
+    }
+
+    public static BitSet listBitSet(final List<Boolean> list) {
+        final int n = list.length();
+        if (n > MAX_BIT_SIZE) {
+            throw new IllegalArgumentException("Does not support lists greater than " + MAX_BIT_SIZE + " bits, actual size is " + n);
+        }
+        long result = 0;
+        for (Boolean b: list) {
+            result = (result << 1) | toInt(b);
+        }
+        return longBitSet(result);
+    }
+
+    public static BitSet streamBitSet(final Stream<Boolean> s) {
+        return listBitSet(s.toList());
+    }
+
+    public static BitSet stringBitSet(final String s) {
+        return streamBitSet(Stream.fromString(s).map(c -> toBoolean(c)));
+    }
+
+    public boolean isSet(final int index) {
+        return (value & (BASE_LONG << index)) != 0;
+    }
+
+    public boolean isEmpty() {
+        return value == 0;
+    }
+
+    public BitSet set(final int index) {
+        return longBitSet(value | (BASE_LONG << index));
+    }
+
+    public BitSet set(final int index, boolean b) {
+        return b ? set(index) : clear(index);
+    }
+
+    public BitSet clear(final int index) {
+        return longBitSet(value & ~(BASE_LONG << index));
+    }
+
+    public long longValue() {
+        return value;
+    }
+
+    public BitSet and(final BitSet bs) {
+        return longBitSet(value & bs.longValue());
+    }
+
+    public BitSet or(final BitSet bs) {
+        return longBitSet(value | bs.longValue());
+    }
+
+    public BitSet shiftRight(final int n) {
+        return longBitSet(value >> n);
+    }
+
+    public BitSet shiftLeft(final int n) {
+        return longBitSet(value << n);
+    }
+
+    public int bitsUsed() {
+        return toStream().length();
+    }
+
+    /**
+     * Returns a stream of boolean where the head is the most significant bit
+     * (the bit with the largest value)
+     */
+    public Stream<Boolean> toStream() {
+        return Stream.fromString(Long.toBinaryString(value)).map(c -> toBoolean(c)).dropWhile(b -> !b);
+    }
+
+    public String toString() {
+        return Show.bitSetShow.showS(this);
+    }
+
+    public boolean equals(Object obj) {
+        return Equal.equals0(BitSet.class, this, obj, () -> Equal.bitSetSequal);
```

Comment:
you can use the overload of ```equals0``` that take directly the ```Equal``` instance as last parameter.